### PR TITLE
New rcparam to set default axes title location

### PR DESCRIPTION
**Pull request by yeosingng**
## PR Summary
UTSC student creating pull request for for CSCD01.
Added new rcparam to set default axes title location requested in #9682.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: matplotlibrc.template [hunk 1]
```diff
@@ -294,6 +294,7 @@
 #axes.grid.axis      : both    ## which axis the grid should apply to
 #axes.grid.which     : major   ## gridlines at major, minor or both ticks
 #axes.titlesize      : large   ## fontsize of the axes title
+#axes.titlelocation  : center  ## alignment of the title
 #axes.titleweight    : normal  ## font weight of title
```
Comment (ImportanceOfBeingErnest) [inline matplotlibrc.template]: Maybe a very minor point here: I wouldn't put the location *in between* two parameters which set some font attributes. Maybe rather put it above `size` or below `weight`? Also the comment could mention the three possible values of this parameter.

Comment (yeosingng) [reply matplotlibrc.template]: I think that's a valid point, I think above size would be a good location

File: lib/matplotlib/rcsetup.py [hunk 2]
```diff
@@ -1180,6 +1180,7 @@ def _validate_linestyle(ls):
 
     'axes.titlesize':        ['large', validate_fontsize],  # fontsize of the
                                                             # axes title
+    'axes.titlelocation':    ['center', validate_string],  # alignment of axes title
```
Comment (ImportanceOfBeingErnest) [inline lib/matplotlib/rcsetup.py]: Maybe the validation should already make sure only valid locations are passed? E.g. adding a 

    validate_titlelocation = ValidateInStrings('axes.titlelocation', ['left', 'center', 'right'])

and using it as

    'axes.titlelocation':    ['center', validate_titlelocation],  # alignment of axes title

**Comment by ImportanceOfBeingErnest** [comment]
In general, this looks good. In addition to the two suggestions above, maybe this should get a test  and a what's new notice?

**Comment by yeosingng** [comment]
I'll work on the suggestions above and a what's new. In terms of testing, what type of test do you suggest for an implementation like this? For title alignment I think there's a test `test_title_location_roundtrip()` in test_axes.py that may cover the cases. Other than the axes itself, are there any standard tests to add when adding a new rcparam?

**Comment by ImportanceOfBeingErnest** [comment]
I think what we want to test is that the location specified by the rcParams is correctly applied to the title. This may sound trivial, but such tests are still useful. It looks like you can simply amend the `test_title_location_roundtrip()` test with a part similar to the existing one, just with the rcParam in use.

**Comment by yeosingng** [comment]
I have modified the existing test by setting the rcParam explicitly in `test_title_location_roundtrip()`, and have implemented everything else suggested, which consists of: moving the order of the param, validation for correct values, and a 'what's new' notice.

**Comment by timhoffm** [comment]
Not quite sure if this should be `axes.titlelocation` or `axes.titleloc`. The former is more clear, while the latter matches the parameter name `set_title(loc='...')`. Probably `axes.titlelocation` is still better, but we should make that decision consciously.

So, as a quick vote:
- :+1: for `axes.titlelocation`
- :-1: for `axes.titleloc`

**Pull request merged**